e2e/pages/homePage.ts

```diff
@@ -0,0 +1,22 @@
+import { Page } from '@playwright/test';
+
+export class HomePage {
+  constructor(readonly page: Page) {}
+
+  readonly patientSearchIcon = () => this.page.locator('[data-testid="searchPatientIcon"]');
```

Comment:
I really dislike the idea of adding testids to production code just for this purpose, but also, if we're already using `locator()`, we should just take advantage of existing markup as much as possible:

```
readonly patientSearchIcon = () => this.page.locator('[name="SearchPatientIcon"]');
readonly patientSearchBar = () => this.page.getByPlaceholder('Search for a patient by name or identifier number');
readonly floatingSearchResultsContainer = () => this.page.getByTestId('floatingSearchResultsContainer'));
```

Reply:
I agree with your dislike of adding data-testid attributes to the production code. However, there are some benefits to using data-testid attributes for e2e testing. One benefit is that data-testid attributes are not coupled to the behavior or styling of an element, which can make them more stable and reliable for testing purposes. Additionally, using data-testid attributes can make it clear to everyone that a specific element is being used directly by test code.

It is generally considered an anti-pattern to rely heavily on other types of selectors, such as class names or element IDs, for e2e testing. This is because these types of selectors can change over time, which can make tests less stable and reliable. Instead, it is generally recommended to use data-testid attributes or other types of dedicated test selectors to identify elements for testing purposes.  ex: https://docs.cypress.io/guides/references/best-practices#Selecting-Elements

As you mentioned, it is considered best practice to remove data-testid attributes from the production version of a frontend application. This can help minimize complexity, improve the user experience, and reduce the risk of security vulnerabilities. One way to do this is to use a tool like webpack-strip to remove data-testid attributes when bundling the source code for production.